src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -137,7 +137,7 @@ cdef void plugin_get_metadata(
   cy_context.context = context
   self.plugin_callback(cy_context, python_callback)
 
-cdef void plugin_destroy_c_plugin_state(void *state):
+cdef void plugin_destroy_c_plugin_state(void *state) with gil:
```

Comment:
Ah. The problem is that Cython correctly sees that, from Python code, this function is only ever called when the GIL is held (due to its old signature implicitly requesting that calling code hold the GIL, else it'd be `nogil`), but we let this function's pointer escape into a non-Python non-Cython function, so the transpiler can't statically enforce that guarantee.